dev/com.ibm.ws.transaction.cdi/src/com/ibm/tx/jta/cdi/TransactionContextExtension.java

```diff
@@ -26,9 +44,83 @@
+    private final AtomicServiceReference<CDIService> cdiServiceRef = new AtomicServiceReference<CDIService>("cdiService");
+
+    private BeanManager beanManager = null;
+
     public void afterBeanDiscovery(@Observes AfterBeanDiscovery event, BeanManager manager) {
-        TransactionContext tc = new TransactionContext();
+        TransactionContext tc = new TransactionContext(getBeanManager());
         com.ibm.tx.jta.impl.TransactionImpl.registerTransactionScopeDestroyer(tc);
-        event.addContext(new TransactionContext());
+        event.addContext(tc);
+    }
+
+
+    private BeanManager getBeanManager() {
+        if (beanManager == null) {
+            BundleContext bundleContext = getBundleContext(FrameworkUtil.getBundle(getClass()));
+            if (bundleContext != null) {
+                ServiceReference<CDIService> serviceReference = getServiceReferences(bundleContext);
+                if (serviceReference != null) {
+                    CDIService cdiService = bundleContext.getService(serviceReference);
+                    if (cdiService != null) {
+                        beanManager = cdiService.getCurrentBeanManager();
+                    }
+                }    
+            }
+        }
+        return beanManager;
+    }
+
+    private static BundleContext getBundleContext(final Bundle bundle) {
+        if (System.getSecurityManager() == null)
+            return bundle.getBundleContext();
+        else
+            return AccessController.doPrivileged(new PrivilegedAction<BundleContext>() {
+                @Override
+                public BundleContext run() {
+                    return bundle.getBundleContext();
+                }
+            });
+    }
+
+    @FFDCIgnore(PrivilegedActionException.class)
+    private static ServiceReference<CDIService> getServiceReferences(final BundleContext bCtx) {
+        final Class<CDIService> cdiServiceClass = CDIService.class;
+        if (System.getSecurityManager() == null) {
+            return bCtx.getServiceReference(cdiServiceClass);
+        } else {
+            try {
+                return AccessController.doPrivileged(new PrivilegedExceptionAction<ServiceReference<CDIService>>() {
+                    @Override
+                    public ServiceReference<CDIService> run() throws InvalidSyntaxException {
+                        return bCtx.getServiceReference(cdiServiceClass);
+                    }
+                });
+            } catch (PrivilegedActionException e) {
+                throw new RuntimeException(e);
+            }
+        }
+    }
+
+    public void activate(ComponentContext context) {
+        cdiServiceRef.activate(context);
+    }
+
+    public void deactivate(ComponentContext context) {
+        cdiServiceRef.deactivate(context);
+    }
+
+    @Reference(name = "cdiService", service = CDIService.class)
+    protected void setCdiService(ServiceReference<CDIService> ref) {
```

Comment:
this method, the unset method, the activate, and the deactivate can be removed and you can instead just use a field-based annotation like this:
```java
@Reference
CDIService cdiSvc;
```